### DISCUSSION TITLE
Default klog file

### DISCUSSION THREAD

**Discussion by ulken** [jotaen/klog#241] [Feature ideas & discussions]
With a `@default` bookmark in place, is there anything hindering a default klog file (at `~/.klog/default.klg` or similar)?

Wouldn't it be a nice getting started experience to just type `klog start` and be good to go?

Might just be me, but I would have guessed many don't use more than one file and it would be nice to not have to care about where it's stored.

It would still be accessible via `klog edit`.

**Comment by jotaen** [comment]
I see your point about reducing friction when getting started. I’m not sure, though, klog would lend itself well to the idea of a default file path. In contrast to other CLI time tracking tools, klog deliberately isn’t opinionated in regards to the storage format or location.

I don’t have comprehensive insights into average usage behaviour, but my assumption would be that it’s more common to have multiple files, as a single file could get quite large at some point. That’s not so much a problem in regards to performance or disk space, but more in regards the convenience to handle it in an editor.

A potential idea could be to add more convenience to the bookmarks feature, e.g. that if you try to use an unset bookmark name, it would offer you to create the file for you, instead of just telling you that the bookmark is not there.

**Reply by ulken** [reply]
OK, no biggie, just a thought. Thanks for your input on the matter.

**Comment by rafalkrupinski** [comment]
Default bookmark, that I could skip from the command line, would be great. I could set it to the current year file.

**Reply by rafalkrupinski** [reply]
BTW, thank you Jan for this awesome project, I love the simple design.

**Reply by rafalkrupinski** [reply]
Wait a minute, default bookmark works like I just described :)

**Reply by jotaen** [reply]
Yeah, I suppose I could clarify the documentation around the default bookmark a little bit, to make it better to understand.

So just for the records here: there is a special bookmark in klog, that allows you to use klog without having to specify any file name. Internally, that bookmark goes by the name `@default`.

You can set up that bookmark by doing e.g.

```
$ klog bookmark set ~/some-path/myfile.klg
```

With that, you can use all regular klog commands without having to specify an input file name, and klog automatically directs these commands to the `@default` bookmark – e.g.:

```
$ klog total
```

```
$ klog start
```

(Although you can still specify input file names as usual.)

> BTW, thank you Jan for this awesome project

Thanks for saying this, glad to hear.
